trieste/acquisition/function.py

```diff
@@ -157,6 +158,133 @@ def expected_improvement(model: ProbabilisticModel, eta: TensorType, at: TensorT
+        Q(q) = a + b * log( -1 * log(1 - q)). We choose values for a and b that
+        match the Gumbel's interquartile range with that of the observed
+        empirical cumulative density function of Pr(y*<y).
+
+        We then sample this Gumbel distribution by sampling from a uniform random variable
+        and applying the inverse probability integral transform, i.e. given a sample
+        r ~ Unif[0,1] then g = a + b * log( -1 * log(1 - r)) follows g ~ Gumbel(a,b).
+
+
+        :param dataset: The data from the observer.
+        :param model: The model over the specified ``dataset``.
+        :return: The MES function.
+        """
+        if len(dataset.query_points) == 0:
+            raise ValueError("Dataset must be populated.")
+        query_points = self._search_space.sample(num_samples=self._grid_size)
+        query_points = tf.concat([dataset.query_points, query_points], 0)
+        fmean, fvar = model.predict(query_points)
+        fsd = tf.math.sqrt(fvar)
+
+        def probf(y: tf.Tensor) -> tf.Tensor:  # Build empirical CDF for Pr(y*^hat<y)
+            unit_normal = tfp.distributions.Normal(tf.cast(0, fmean.dtype), tf.cast(1, fmean.dtype))
+            log_cdf = unit_normal.log_cdf(-(y - fmean) / fsd)
+            return 1 - tf.exp(tf.reduce_sum(log_cdf, axis=0))
+
+        left = tf.reduce_min(fmean - 5 * fsd)
+        right = tf.reduce_max(fmean + 5 * fsd)
+
+        def binary_search(val: float) -> float:  # Find empirical interquartile range
+            return bisect(lambda y: probf(y) - val, left, right, maxiter=10000)
+
+        q1, q2 = map(binary_search, [0.25, 0.75])
+
+        log = tf.math.log
+        l1 = log(log(4.0 / 3.0))
+        l2 = log(log(4.0))
+        b = (q1 - q2) / (l1 - l2)
+        a = (q2 * l1 - q1 * l2) / (l1 - l2)
+
+        uniform_samples = tf.random.uniform([self._num_samples], dtype=fmean.dtype)
+        self.gumbel_samples = log(-log(1 - uniform_samples)) * tf.cast(b, fmean.dtype) + tf.cast(
+            a, fmean.dtype
+        )
+
+        return lambda at: self._acquisition_function(model, self.gumbel_samples, at)
+
+    @staticmethod
+    def _acquisition_function(
+        model: ProbabilisticModel, samples: TensorType, at: TensorType
+    ) -> TensorType:
+        return min_value_entropy_search(model, samples, at)
+
+
+def min_value_entropy_search(
+    model: ProbabilisticModel, samples: TensorType, at: TensorType
+) -> TensorType:
+    r"""
+    Computes the information gain, i.e the change in entropy of p_min (the distribution of the
+    minimal value of the objective function) if we would evaluate x.
+
+    See the following for details:
+
+    ::
+
+        @article{wang2017max,
+          title={Max-value entropy search for efficient Bayesian optimization},
+          author={Wang, Zi and Jegelka, Stefanie},
+          journal={arXiv preprint arXiv:1703.01968},
+          year={2017}
+        }
```

Comment:
since we're here, can we convert this to the sphinxcontrib-bibtex format? That would just mean moving the reference to docs/refs.bib and using
```
See :cite:`wang2017max` for details.
```
here